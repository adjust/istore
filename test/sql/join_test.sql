BEGIN;
-- join join two tables;
-- ./spec/join_spec.rb:27;
CREATE EXTENSION istore;
SELECT os_name, device_type, country, a.num+b.num FROM(
VALUES
('android'::os_name, 'phone'::device_type, 'de'::country, 10),
('android', 'phone', 'us', 10),
('android', 'phone', 'de', 10)
)a(os_name, device_type, country, num)
JOIN (
VALUES
('android'::os_name, 'phone'::device_type, 'de'::country, 10),
('android', 'phone', 'us', 10),
('android', 'phone', 'de', 10),
('windows', 'phone', 'de', 10)
)b(os_name, device_type, country, num)
USING (os_name, device_type, country)
;
ROLLBACK;
