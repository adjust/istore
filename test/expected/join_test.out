BEGIN;
-- join join two tables;
-- ./spec/join_spec.rb:27;
CREATE EXTENSION istore;
SELECT os_name, device_type, country, a.num+b.num FROM(
VALUES
('android'::os_name, 'phone'::device_type, 'de'::country, 10),
('android', 'phone', 'us', 10),
('android', 'phone', 'de', 10)
)a(os_name, device_type, country, num)
JOIN (
VALUES
('android'::os_name, 'phone'::device_type, 'de'::country, 10),
('android', 'phone', 'us', 10),
('android', 'phone', 'de', 10),
('windows', 'phone', 'de', 10)
)b(os_name, device_type, country, num)
USING (os_name, device_type, country)
;
 os_name | device_type | country | ?column? 
---------+-------------+---------+----------
 android | phone       | de      |       20
 android | phone       | de      |       20
 android | phone       | us      |       20
 android | phone       | de      |       20
 android | phone       | de      |       20
(5 rows)

ROLLBACK;
